### DISCUSSION TITLE
code: 'MODULE_NOT_FOUND', error on running Artillery script

### DISCUSSION THREAD

**Discussion by dtbnakum** [artilleryio/artillery#2641] [Artillery]
I am running a simple login test but while execution getting code: 'MODULE_NOT_FOUND'. I checked my Artillery installation it seems proper. Not sure what's wrong with my test. Could you please help?

PS C:\Users\Bhavesh.Nakum\OneDrive - DynaTech Systems Pvt. Ltd\Documents\Test Automation\AtrilleryPlaywright> artillery run artillery.yml
Test run id: tmwan_ezbfeg4frdptjcycg9a7fr4repzhr_5adg
/ worker error, id: 1 Error: Cannot find module 'C:\Users\Bhavesh.Nakum\OneDrive - DynaTech Systems Pvt. Ltd\Documents\Test Automation\AtrilleryPlaywright\tests\artillery\login.js'
Require stack:
- C:\Users\Bhavesh.Nakum\AppData\Roaming\npm\node_modules\artillery\node_modules\@artilleryio\int-core\lib\runner.js   
- C:\Users\Bhavesh.Nakum\AppData\Roaming\npm\node_modules\artillery\node_modules\@artilleryio\int-core\index.js        
- C:\Users\Bhavesh.Nakum\AppData\Roaming\npm\node_modules\artillery\lib\artillery-global.js
- C:\Users\Bhavesh.Nakum\AppData\Roaming\npm\node_modules\artillery\lib\platform\local\worker.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at loadProcessor (C:\Users\Bhavesh.Nakum\AppData\Roaming\npm\node_modules\artillery\node_modules\@artilleryio\int-core\lib\runner.js:97:33)
    at prepare (C:\Users\Bhavesh.Nakum\AppData\Roaming\npm\node_modules\artillery\lib\platform\local\worker.js:109:24) 
    at async MessagePort.onMessage (C:\Users\Bhavesh.Nakum\AppData\Roaming\npm\node_modules\artillery\lib\platform\local\worker.js:50:5) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\Bhavesh.Nakum\\AppData\\Roaming\\npm\\node_modules\\artillery\\node_modules\\@artilleryio\\int-core\\lib\\runner.js',
    'C:\\Users\\Bhavesh.Nakum\\AppData\\Roaming\\npm\\node_modules\\artillery\\node_modules\\@artilleryio\\int-core\\index.js',
    'C:\\Users\\Bhavesh.Nakum\\AppData\\Roaming\\npm\\node_modules\\artillery\\lib\\artillery-global.js',
    'C:\\Users\\Bhavesh.Nakum\\AppData\\Roaming\\npm\\node_modules\\artillery\\lib\\platform\\local\\worker.js'        
  ]
}
PS C:\Users\Bhavesh.Nakum\OneDrive - DynaTech Systems Pvt. Ltd\Documents\Test Automation\AtrilleryPlaywright>

**Comment by bernardobridge** [comment]
Hi @dtbnakum 👋 ,

Can you please share:
- The contents of your YAML file (`artillery.yml`)
- The contents of your `login.js` file
- The directory structure (i.e. is the `login.js` file in the same location as `artillery.yml`?)

If possible, please use [Github Code Blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) when sharing the file contents, to improve readability. Thanks!

**Reply by dtbnakum** [reply]
@bernardobridge  I got what was issue, my file name was different than login.js which I mentioned in .yml file for processor. my bad. Now that error is resolved. 

However, vusers are failing and I don't see any errors in console.

vusers.created: ................................................................ 1
vusers.created_by_name.SAICALogin: ............................................. 1
vusers.failed: ................................................................. 1

**Reply by bernardobridge** [reply]
Hey @dtbnakum 👋 

Yes, there's a current bug affecting that. We will have a new release out that fixes it later today. I'll update you once it's out.

**Reply by bernardobridge** [reply]
@dtbnakum : [Artillery v2.0.10](https://github.com/artilleryio/artillery/releases/tag/artillery-2.0.10) is out containing a fix for this. If you update, you should start seeing errors.

**Reply by dtbnakum** [reply]
@bernardobridge Thank you for quick response. I will update version and share details if required any further help on this.

**Reply by dtbnakum** [reply]
@bernardobridge I have updated to latest version v2.0.10 and after running test getting error "**TypeError: fn is not a function**"

Please refer below error details and attached index.js as zip file for reference
[index.zip](https://github.com/artilleryio/artillery/files/14987109/index.zip)


- TypeError: fn is not a function
    at Array.scenario (C:\Users\Bhavesh.Nakum\AppData\Roaming\npm\node_modules\artillery\node_modules\artillery-engine-playwright\index.js:340:17)
    at runNextTicks (node:internal/process/task_queues:60:5)
    at process.processImmediate (node:internal/timers:449:9)
--------------------------------------
Metrics for period to: 10:42:10(+0530) (width: 1.347s)
--------------------------------------

vusers.created: ................................................................ 1
vusers.created_by_name.0: ...................................................... 1
vusers.failed: ................................................................. 1


Warning: multiple batches of metrics for period 1713244320000 2024-04-16T05:12:00.000Z
All VUs finished. Total time: 3 seconds

--------------------------------
Summary report @ 10:42:09(+0530)
--------------------------------

vusers.created: ................................................................ 1
vusers.created_by_name.0: ...................................................... 1
vusers.failed: ................................................................. 1
Log file: test-run-report.json

**Reply by bernardobridge** [reply]
Hi @dtbnakum 👋 

The zip file you shared is just Artillery's source code. Instead, please share the scenario YAML you are using (⚠️ using [Github Code Blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)), as well as your processor file (JS file you're using), so I can help troubleshoot what might be wrong.

**Comment by dtbnakum** [comment]
@bernardobridge PFA 
[requested details.zip](https://github.com/artilleryio/artillery/files/14995005/requested.details.zip)

**Reply by bernardobridge** [reply]
Hi @dtbnakum 👋 . Thanks, I'll have a look soon to see what might be wrong. 👍

**Reply by bernardobridge** [reply]
Hi @dtbnakum . There are two things wrong with your script:

- Your config has `processors` (plural), but the key should be `processor` (singular)
- The path you specified (`./tests/artillery/login.js`). It should be `./artillery/login.js`, as the path should be relative to the location of the config.

**Reply by dtbnakum** [reply]
@bernardobridge  Thank you, changing processors (plural) to processor (singular) resolved issue. path was correct in the config. Now working fine. Appreciate your quick responses and help!
